### PR TITLE
Fix usage of drop

### LINKED ISSUE #778: GH actions checks complain about "calls to `std::mem::drop` with a reference instead of an owned value does nothing"
Looking at any recent pull request, the "check-juliaup" action reports several warnings about
> calls to `std::mem::drop` with a reference instead of an owned value does nothing

as "Unchanged files with check annotations". See e.g. [here](https://github.com/JuliaLang/juliaup/pull/753/files) (that PR is *not* causing those warnings, they come from the `main` branch).

Unfortunately I don't really know Rust and so can't judge whether these are problematic, or what the correct fix (if any) is, but it would be nice to not have this noise show up in all PRs.

### PR DESCRIPTION
**Pull request by fingolfin**
Fixes #778

### CODE REVIEW

**Comment by davidanthoff** [comment]
@fingolfin Did you test this PR by adding and then removing a cron job? I don't have a Linux machine, but only use WSL, but that uses a different code-path and won't hit this...

**Comment by fingolfin** [comment]
No I did not, as I do not know how to use this feature. I could not find any documentation for it.

**Comment by fingolfin** [comment]
If you can give me a pointer, I am happy to test it on a Linux machine.

**Comment by davidanthoff** [comment]
Run `juliaup config backgroundselfupdateinterval 5` to configure a cron job that looks for a new Juliaup version every 5 minutes. `juliaup config backgroundselfupdateinterval 0` should uninstall the cron job again. I think that should hit this code path, but if you want to be extra careful you could add some kind `print` statement to make sure that is really so :)

**Comment by fingolfin** [comment]
OK those work with my regular (release) juliaup. But I can't get it to work with a debug build made from this repository. Specifically, `juliaup config` does not know about `backgroundselfupdateinterval` ?!

I am probably doing something stupid (I am a rust & cargo n00b). Here is what I am trying (on macOS):
```
$ cargo b
...  yadda yadda ...
$ cargo r --bin juliaup config --help
warning: file `/Users/mhorn/Projekte/Julia/packages/JuliaLang/juliaup/src/bin/julialauncher.rs` found to be present in multiple build targets:
  * `bin` target `julia`
  * `bin` target `julialauncher`
    Finished dev [unoptimized + debuginfo] target(s) in 0.32s
     Running `target/debug/juliaup config --help`
Juliaup configuration

Usage: juliaup config <COMMAND>

Commands:
  channelsymlinks           Create a separate symlink per channel
  versionsdbupdateinterval  The time between automatic updates of the versions database in minutes, use 0 to disable
  help                      Print this message or the help of the given subcommand(s)

Options:
  -h, --help  Print help
```

**Comment by davidanthoff** [comment]
You need to add the `selfupdate` feature, does `cargo run --features selfupdate --bin juliaup -- config --help` work?

**Comment by fingolfin** [comment]
tl;dr: I confirm this branch works.

Ahhh, I did not know about "features", gotta read up on that. But for now I tried what you suggested and it works for `help`. However, setting the selfupdate interval still fails with an error that it can't open "juliaup config file" -- but also in `main`, i.e. not specific to this branch. I get this:
```
...
    Finished dev [unoptimized + debuginfo] target(s) in 0.13s
     Running `target/debug/juliaup config backgroundselfupdateinterval 5`
Error: `config` command failed to load configuration data.

Caused by:
    0: Failed to open juliaup config file.
    1: No such file or directory (os error 2)
```
That's not very helpful. But I next tried the `status` command and that gave a bit more helpful output (still on `main` branch):
```
$ cargo run --features selfupdate --bin juliaup -- status
warning: file `/Users/mhorn/Projekte/Julia/packages/JuliaLang/juliaup/src/bin/julialauncher.rs` found to be present in multiple build targets:
  * `bin` target `julia`
  * `bin` target `julialauncher`
warning: calls to `std::mem::drop` with a reference instead of an owned value does nothing
   --> src/operations.rs:540:13
    |
540 |             drop(child_stdin);
    |             ^^^^^-----------^
    |                  |
    |                  argument has type `&mut ChildStdin`
    |
    = note: use `let _ = ...` to ignore the expression or result
    = note: `#[warn(dropping_references)]` on by default

warning: calls to `std::mem::drop` with a reference instead of an owned value does nothing
   --> src/operations.rs:589:13
    |
589 |             drop(child_stdin);
    |             ^^^^^-----------^
    |                  |
    |                  argument has type `&mut ChildStdin`
    |
    = note: use `let _ = ...` to ignore the expression or result

warning: `juliaup` (lib) generated 2 warnings
    Finished dev [unoptimized + debuginfo] target(s) in 0.12s
     Running `target/debug/juliaup status`
Error: `status` command failed to load configuration file.

Caused by:
    Could not open self configuration file "/Users/mhorn/Projekte/Julia/packages/JuliaLang/juliaup/target/juliaupself.json": Os { code: 2, kind: NotFound, message: "No such file or directory" }
```
So now at least I know which file it tried to open (perhaps the other error messages of this kind could all be made to include this path?).

I guess this file is normally created by the juliaup installer. Anyway, I simply copied the one from my regular juliaup config:

    cp ~/.julia/juliaup/juliaupself.json target/

That seemed to do the trick, and with all that, I could now test the feature *with this branch*:
```
$ crontab -l  # outputs nothing, crontab is empty


$ cargo run --features selfupdate --bin juliaup -- config backgroundselfupdateinterval 5
warning: file `/Users/mhorn/Projekte/Julia/packages/JuliaLang/juliaup/src/bin/julialauncher.rs` found to be present in multiple build targets:
  * `bin` target `julia`
  * `bin` target `julialauncher`
    Finished dev [unoptimized + debuginfo] target(s) in 0.13s
     Running `target/debug/juliaup config backgroundselfupdateinterval 5`
Property 'backgroundselfupdateinterval' set to '5'


$ crontab -l
*/5 * * * * /Users/mhorn/Projekte/Julia/packages/JuliaLang/juliaup/target/debug/juliaup 4c79c12db1d34bbbab1f6c6f838f423f


$ cargo run --features selfupdate --bin juliaup -- config backgroundselfupdateinterval 0
warning: file `/Users/mhorn/Projekte/Julia/packages/JuliaLang/juliaup/src/bin/julialauncher.rs` found to be present in multiple build targets:
  * `bin` target `julia`
  * `bin` target `julialauncher`
    Finished dev [unoptimized + debuginfo] target(s) in 0.12s
     Running `target/debug/juliaup config backgroundselfupdateinterval 0`
Property 'backgroundselfupdateinterval' set to '0'


$ crontab -l  # again outputs nothing
```
